lib/moon/design/button.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Moon.Design.Button do
+  @moduledoc false
+
+  use Moon.StatelessComponent
+
+  alias Moon.Design.Button.Utils
+  alias Moon.Icon
+  alias Moon.Components.Loader
+
+  prop(variant, :string,
+    values: ["primary", "secondary", "tertiary", "ghost"],
+    default: "primary"
+  )
+
+  prop(size, :string, values: ["xs", "sm", "md", "lg", "xl"], default: "md")
+  prop(as, :string, values!: ~w(a button), default: "button")
+  prop(full_width, :boolean)
+  prop(disabled, :boolean)
+  prop(animation, :string, values: ~w(progress success error pulse))
+  prop(class, :css_class)
+
+  prop(left_icon, :string)
+  prop(right_icon, :string)
+
+  slot(default)
+  slot(right_icon_slot)
+  slot(left_icon_slot)
+
+  prop(button_gap, :css_class, default: "gap-2")
+
+  def render(assigns) do
+    ~F"""
+    <button
+      {=@as}
+      class={
+        "flex row justify-center items-center gap-2 relative overflow-hidden active:scale-90 transition-all font-semibold",
+        @class,
+        Utils.get_button_size_classes(@size),
+        "#{Utils.get_no_icon_padding(@size)}": !@right_icon && !@left_icon,
+        "#{Utils.get_right_icon_paddings(@size, @right_icon)}": @right_icon && !@full_width,
```

Comment:
u already do a check outside a function, so no second parameter should be given to a function, just `@size` 
same for the functions get_left_icon_paddings & get_full_width_padding